analyses/fusion_filtering/01-fusion-standardization.R

```diff
@@ -0,0 +1,90 @@
+#		annots=Annotation from FusionAnnotator] are appended to the 
+#		right of original arriba data frame 
+#
+# caller :	Caller used for the input merged fusion calls [STARfusion|| Arriba]
+#
+
+library("optparse")
+
+option_list <- list( 
+     make_option(c("-f", "--fusionfile"),type="character",
+        help="Merged fusion calls from [STARfusion || Arriba]"),
+     make_option(c("-c", "--caller"), type="character", 
+        help="Caller type [STARfusion|| Arriba]"),
+     make_option(c("-o","--outputfile"),type="character",
+	help="Standardized fusion calls from [STARfusion || Arriba] (.RDS)")	
+    )
+
+# get command line options, if help option encountered print help and exit,
+
+opt <- parse_args(OptionParser(option_list=option_list))
+
+inputfile<-opt$fusionfile
+caller<-opt$caller
+outputfile<-opt$outputfile
+
+fusion_calls<-read.delim(opt$fusionfile,stringsAsFactors=FALSE)
+#To have a general column with unique IDs associated with each sample
+fusion_calls$Sample<-fusion_calls$tumor_id
+
+
+
+standard_fusion <- function(fusion_calls=fusion_calls,caller=caller) {
+# @param fusion_calls A dataframe from star fusion or arriba (more callers to be added)
+# @param caller string options STARfusion/arriba
+# @return Standardized fusion calls ready for filtering
+  
+  if(caller=="STARfusion"){
+  	fusion_calls$LeftBreakpoint <- gsub('^chr','',fusion_calls$LeftBreakpoint)
+  	fusion_calls$RightBreakpoint <- gsub('^chr','',fusion_calls$RightBreakpoint)
+  	fusion_calls$Fusion_Type<-fusion_calls$PROT_FUSION_TYPE
+  	fusion_calls$Fusion_Type<-gsub("INFRAME","in-frame",fusion_calls$Fusion_Type)
+  	fusion_calls$Fusion_Type<-gsub("FRAMESHIFT","frameshift",fusion_calls$Fusion_Type)
+  	fusion_calls$Fusion_Type[-which(fusion_calls$Fusion_Type %in% c("in-frame","frameshift"))] <- 'other'
+  	fusion_calls$Caller <- 'STARFusion'
+  	fusion_calls$FusionName<-fusion_calls$FusionName
+  	fusion_calls$Confidence<-"NA"
+  	standard_calls <- unique(fusion_calls[,c('FusionName','Sample','Caller','Fusion_Type','JunctionReadCount','SpanningFragCount','Confidence','annots')])
+  	return(standard_calls)
+
+  }
+  if( caller=="Arriba"){
+	fusion_calls$LeftBreakpoint <- gsub('^chr','',fusion_calls$breakpoint1)
+    	fusion_calls$RightBreakpoint <- gsub('^chr','',fusion_calls$breakpoint2)
+    	fusion_calls$Fusion_Type<-fusion_calls$reading_frame
+    	fusion_calls$Fusion_Type<-gsub("out-of-frame","frameshift",fusion_calls$Fusion_Type)
+    	fusion_calls$Fusion_Type[-which(fusion_calls$Fusion_Type %in% c("in-frame","frameshift"))] <- 'other'
+    	fusion_calls$Caller <- 'arriba'
+    	fusion_calls$FusionName <-paste0(gsub(",","/",fusion_calls$gene1),"--",gsub(",","/",fusion_calls$gene2))
+    	fusion_calls$JunctionReadCount<-fusion_calls$split_reads1+fusion_calls$split_reads2
+    	fusion_calls$SpanningFragCount<-fusion_calls$discordant_mates
+    	fusion_calls$Confidence<-fusion_calls$confidence
+	colnames(fusion_calls)[27]<-"annots"
```

Comment:
Yeah soo, I've added the 2 columns in arriba from FusionAnnotator output after the actual Arriba dataframe+"tumor_id" (and would expect the user to format the calls the same way to have as input) . But you are right, users might prefer to add more columns before running FusionAnnotator or vise versa. I'll use the ncol(fusion_calls) suggested but so it's clear to have the annotation as the last column of the dataframe.

Reply:
So I fixed it with colnames(fusion_calls)[which(colnames(fusion_calls)=="X..")] <- "annots" since the ncol(fusion_calls) was not annots because I add "Sample" column to the dataframe in the script. . I think I should fix the arriba inputfile itself to have a column "annots" instead of "X.." from my FusionAnnotator run

Reply:
> I think I should fix the arriba inputfile itself to have a column "annots" instead of "X.." from my FusionAnnotator run

Agree that this seems like the best course of action. The `X..` column name is part of the Arriba file that is part of `release-v4-20190909`, is that correct? Can this change be included in the next release? cc: @jharenza

Reply:
@jaclyn-taroni - I would normally agree with this, however, if I am understanding this correctly, the `X..` is standard output from the FusionAnnotator run (is that correct, @kgaonkar6?). FusionAnnotator is standard in STAR-Fusion, but not Arriba. Initially that step was in this fusion pipeline, but for the package and the sake of the large database download and time for annotation, we decided to run this prior to starting the package scripts and require this be run by users prior to data input into the package. 

1. Do you agree with requiring this be run prior to the package input, @jaclyn-taroni? One caveat with this being a dependency, I think, would be for @kgaonkar6 to either fork this repo (not sure if on github) so that users will always use the same version or archive it ourselves and direct users to that version so it doesn't disappear later.
2. If so, if this colname is default from FusionAnnotator outout, and we require users to run this and use it directly as input to our package, then we should probably change that colname within our code, not have users do this and risk more errrors going in, right?

Reply:
Let's back up for a moment -- what is the command for how FusionAnnotator is run and what does the output from FusionAnnotator look like when it's run for Arriba results? 

A comment here that:
* Includes the command for how FusionAnnotator is run from the command line 
* Includes the first few lines in a code block and then some additional description 

would be great. 

I'm asking because if there is a step where one binds the columns of the Arriba output to the FusionAnnotator output, that could be an opportunity to resolve the column name issue.

Reply:
I am not certain but assuming the output was exactly what we provided in our data release - @kgoankar6 can answer this!

Reply:
bash commands to run format arriba calls and run FusionAnnotator

`# get geneA--geneB format that FusionAnnotator needs
cat pbta_arriba_genefusion.tsv |awk '{ print $1,"--",$2}'|sed 's/ //g' > pbta_arriba_genefusion_list.tsv
# add geneA--geneB to arriba fusion calls to have the dataframe annotated
paste pbta_arriba_genefusion.tsv pbta_arriba_genefusion_list.tsv >pbta_arriba_genefusion_v2.tsv
# run FusionAnnotator 
./FusionAnnotator --genome_lib_dir ../../../references/GRCh38_v27_CTAT_lib_Feb092018/ctat_genome_lib_build_dir/ --annotate ../../../data/interim/pbta_arriba_genefusion_v2.tsv --fusion_name_col 25 > ../../../data/interim/pbta_arriba_genefusion_CTAT.tsv`

Reply:
https://drive.google.com/open?id=1gPeYPKWg4cicSkYQGoGMHQTY-qKOhNlc first few lines of the output from the above commands. The columnname that FusionAnnotator actually provides in  "[ ]". Reading this in R changes the column name to X..

Reply:
> The columnname that FusionAnnotator actually provides in "[ ]". Reading this in R changes the column name to "X.."

In that case, if you can not provide an argument to FusionAnnotator or make the change easily in another shell command, I would leave this as is with the `rename` step.

Reply:
> I'm asking because if there is a step where one binds the columns of the Arriba output to the FusionAnnotator output, that could be an opportunity to resolve the column name issue.

So I don't merge the arriba output with FusionAnnotator output instead I format the arriba output to run FusionAnnotator on the whole file specifying that the column number that has the fusion naming convention that FusionAnnotator requires. Me and @jharenza  digged into the code a little bit and it seems that the fusion name header line should start with "#" to get the "annots" column name from the FusionAnnotator. https://github.com/FusionAnnotator/FusionAnnotator/blob/cee6b3aca14fe86a0ad6be4d468dc8ffb45c0116/FusionAnnotator#L121 

I'll have the file updated in the next release to get the "annots" column name